api/api/validators/generate_teams_validator.py

```diff
@@ -7,23 +14,59 @@ def validate(self):
         self.validate_algorithm_type()
         self.validate_team_options()
         self.validate_algorithm_options()
-        self.validate_projects()
+        self.validate_algorithm_configs()
         self.validate_students()
 
     def _validate_schema(self):
-        pass
+        Schema(dict).validate(self.data)
+        Schema(list).validate(self.data.get("students"))
+        Schema(dict).validate(self.data.get("algorithm_options"))
```

Comment:
this PR is doing a lot of things and I feel like some are falling through the cracks -> this structure is more complex than just being a dictionary

And others too -> students is not just a list of anything and so on and so on, if you're choosing to do all the vlaidation in this PR, it needs to be done properly